prow/lib.sh

```diff
@@ -261,6 +262,49 @@ function connect_kind_clusters() {
   docker exec "${C2_NODE}" ip route add "${C1_POD_CIDR}" via "${C1_DOCKER_IP}"
 }
 
+function install_metallb() {
+  KUBECONFIG="${1}"
+  kubectl apply --kubeconfig="$KUBECONFIG" -f https://raw.githubusercontent.com/metallb/metallb/v0.9.3/manifests/namespace.yaml
+  kubectl apply --kubeconfig="$KUBECONFIG" -f https://raw.githubusercontent.com/metallb/metallb/v0.9.3/manifests/metallb.yaml
+
+  if [ -z "${METALLB_IPS[*]}" ]; then
+    # Take IPs from the end of the docker bridge network subnet to use for MetalLB IPs
+    DOCKER_BRIDGE_SUBNET="$(docker inspect bridge | jq .[0].IPAM.Config[0].Subnet -r)"
```

Comment:
I think this will break when multiple Prow containers run it on the same host. Is this a concern? Not sure how the default bridge network behaves with Docker in docker (in docker)

Reply:
@stevenctl we may have multiple multicluster jobs running on the same host, so if that will break we may need a workaround if possible. Based on resource requests we can have up to 3 jobs per node generally (this may change, just the current state), so the odds of overlapping jobs isn't that low

Reply:
I think putting the kind clusters on their own network would make this much safer.

Reply:
kind doesn't support configuring a docker network, looks like they've planned it though: https://github.com/kubernetes-sigs/kind/issues/273.
Furthermore, this will require changes when we upgrade kind if this is included in the next verison:  
https://github.com/kubernetes-sigs/kind/pull/1508; they use their own network named `kind` instead of `bridge`

Reply:
This is only problematic if the docker-in-docker is setup in a non-isolated way (mounting `/var/run/docker.sock`). If we have a better DinD setup then I think this is fine.

Reply:
> I think this will break when multiple Prow containers run it on the same host. Is this a concern? Not sure how the default bridge network behaves with Docker in docker (in docker)

they're in different network namespaces if you're doing docker in docker.

---

Yep, kind *will* be using a network in v0.8.0, it's just fixed to `kind` unless you experimentally override it. Unfortunately there's no way to get some functionality on the default bridge. 
Conversely we're not quite prepared for wildly custom networks which can be broken in various ways.

You should be fine if you patch this to `kind` when you ugprade. A single network should be fairly desirable for multi-cluster. It's basically equivilant to now just instead of docker's default it's kind's default (since there's no standard network with embedded DNS otherwise, and we fixed a LOT of problems by working with that).